spec/prawn/images_spec.rb

```diff
@@ -43,11 +43,36 @@
     expect(info.height).to eq(453)
   end
 
+  it 'does not close passed-in IO objects' do
+    file = File.open(filename, 'rb')
+    _info = pdf.image(file)
+
+    expect(file).to_not be_closed
+  end
+
   it 'accepts Pathname objects' do
     info = pdf.image(Pathname.new(filename))
     expect(info.height).to eq(453)
   end
 
+  context 'with File message spy' do
```

Comment:
This spec feels a bit too coupled to the implementation and doesn't describe the actual intent particularly well but since I can't give any useful advice on how to improve it, let's keep it. However, tag it with issue. Here's an [example](https://github.com/prawnpdf/prawn/blob/2cdd2e7994c742c97f104e2927d2e2fd3dedc767/spec/prawn/view_spec.rb#L48).

Reply:
I agree, I don't like testing the implementation either. Something like the version with `ObjectSpace` but working on every platform would've been awesome...